openedx/features/enterprise_support/api.py

```diff
@@ -422,8 +422,13 @@ def enterprise_customer_from_session(request):
     """
     Retrieve enterprise_customer data from the request's session,
     returning a ``__CACHE_MISS__`` if absent.
+
+    Now checks for session existence before attempting to access it.
     """
-    return request.session.get(ENTERPRISE_CUSTOMER_KEY_NAME, _CACHE_MISS)
+    if not request or not request.session:
```

Comment:
If the request exists and has no session attribute, this would still throw an uncaught AttributeError. I think you need a `hasattr()` check here instead.